### DISCUSSION TITLE
Any good link available to integrate with Jenkins pipeline for nx-dotnet?

### DISCUSSION THREAD

**Discussion by amitkpal83** [nx-dotnet/nx-dotnet#607] [General]
Any good link available to integrate with Jenkins pipeline for nx-dotnet?

**Comment by bcallaghan-et** [comment]
What sort of integration are you looking for? I use Jenkins to build my mostly-dotnet Nx workspace, and it works fine for me. Notably, my Jenkinsfile has very little reference to .NET or nx-dotnet. For the most part, I just execute the `nx affected` commands relevant to my project.

**Reply by amitkpal83** [reply]
@bcallaghan-et Can you please share your jenkinsfile for reference if you don't mind?

**Reply by bcallaghan-et** [reply]
Unfortunately, that project is proprietary and closed-source. What I can say is that the build pipeline executes the following shell commands, with little modification.
```sh
npm ci
npx nx affected:build --prod
npx nx affected:lint --prod
npx nx affected:test --prod
```

I also set the environment variable `NX_BASE` using the following function
```groovy
def getChangeTarget() {
  // Build against the target of the current pull request, or the last successful build on this branch, or main (in that order).
  return env.CHANGE_TARGET ? 'origin/' + env.CHANGE_TARGET : (env.GIT_PREVIOUS_SUCCESSFUL_COMMIT ?: 'origin/main')
}
```

Note that there is nothing specific to nx-dotnet in these blocks. It's just Nx controlling everything.
